narwhals/dataframe.py

```diff
@@ -186,17 +186,31 @@ def join(
         how: Literal["inner", "left", "cross", "semi", "anti"] = "inner",
         left_on: str | list[str] | None = None,
         right_on: str | list[str] | None = None,
+        on: str | list[str] | None = None,
     ) -> Self:
         _supported_joins = ("inner", "left", "cross", "anti", "semi")
 
         if how not in _supported_joins:
             msg = f"Only the following join strategies are supported: {_supported_joins}; found '{how}'."
             raise NotImplementedError(msg)
 
-        if how == "cross" and (left_on or right_on):
-            msg = "Can not pass left_on, right_on for cross join"
+        if how == "cross" and (left_on or right_on or on):
```

Comment:
seeing as we're changing this, shall we take the chance to be more explicit and write
```
(left_on is not None or right_on is not None or on is not None)
```
?